cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -100,20 +203,99 @@ public static void add(Network n, CgmesModel cgmes) {
                 tapSteps.add(p);
             }
         }
-        cgmes.add(CgmesSubset.STATE_VARIABLES, "SvTapStep", tapSteps);
+
+        for (ThreeWindingsTransformer t : network.getThreeWindingsTransformers()) {
+            PropertyBag p = new PropertyBag(svTapStepProperties);
+            Arrays.asList(t.getLeg1(), t.getLeg2(), t.getLeg3()).forEach(leg -> {
+                if (leg.getPhaseTapChanger() != null) {
+                    p.put(tapChangerPositionName, is(leg.getPhaseTapChanger().getTapPosition()));
+                    p.put(CgmesNames.TAP_CHANGER, cgmes.phaseTapChangerForPowerTransformer(t.getId()));
+                    tapSteps.add(p);
+                } else if (leg.getRatioTapChanger() != null) {
+                    p.put(tapChangerPositionName, is(leg.getRatioTapChanger().getTapPosition()));
+                    p.put(CgmesNames.TAP_CHANGER, cgmes.ratioTapChangerForPowerTransformer(t.getId()));
+                    tapSteps.add(p);
+                }
+            });
+        }
+
+        cgmes.add(originalSVcontext, "SvTapStep", tapSteps);
+    }
+
+    private void addStatusToCgmes() {
+        // create SvStatus, iterate on Connectables, check Terminal status, add
+        // to SvStatus
+        PropertyBags svStatus = new PropertyBags();
+        network.getVoltageLevelStream()
+            .flatMap(VoltageLevel::getConnectableStream)
+            .distinct()
+            .forEach(c -> {
+                PropertyBag p = new PropertyBag(SV_SVSTATUS_PROPERTIES);
+                for (Terminal t : ((Connectable<?>) c).getTerminals()) {
+                    p.put(IN_SERVICE, Boolean.toString(t.isConnected()));
```

Comment:
Not sure this is OK: If a line is connected on side 1 but disconnected on side 2, I'm not sure the IN_SERVICE should be true.

Reply:
@MioRtia @zamarrenolm What do you think?

Reply:
I agree with @mathbagu, not sure what is put in the property bag, it always is the connectivity status of the last terminal, right? Is it what we want?

Reply:
The definition for `inService` attribute for a `ConductingEquipment` states that it is: _The in service status as a result of topology processing_.

For me: if a conducting equipment with multiple terminals has at least one terminal connected, it should be considered in-service. At least it makes sense from a modeling point of view, and its impact in the network could be calculated.

Reply:
In fact Miora seems to say that we only keep the status of the last terminal we have checked. So what is the good solution to put ````inService```` when at least one terminal is connected ?

Reply:
We will change it for "any" terminal, instead of "last"

Reply:
Done